Dark view increases flash damage to eyes

Tayyyyyyy opened the pull request:
Just throwing this out there to go along with #8468 and #8489. I am not sure at all if this is the most balanced way to do this. Unlike nightvision glasses, this just increases the damage done to you by flashing, rather than lowering flash protection, so you'll be fine with sunglasses/welding helmet/etc.

People with dark_view 2 or less won't get any additional damage.

I removed the RNG aspect of flash damage and converted it to using a multiplier instead. If fractional eye damage is bad, I can find a way around this.

1st tier flash eye damage (default vanilla eyes without any protection/antiprotection vs a normal flash (flashbang, handheld flash, cell flash) is dependent on darkness level as well as dark view. Here's a graph, with 0 being completely dark and 10 being fully lit:

![capture](https://user-images.githubusercontent.com/26497062/34464574-66ffd93c-ee42-11e7-8cbe-aae1c77162ed.PNG)


Other ideas:
- Lowering eye protection like night vision glasses
- Increasing stun/confusion length (this will probably require refactoring/snowflake)
- Differing multipliers on extra dark_view eye damage based on eye protection

:cl: Tayyyyyyy, FPK
tweak: The more dark view you have, the more eye damage you take from flashing. Darkness amplifies this effect.
/:cl:

Aurorablade commented:
The issue i have here, and i have seen this before i had to utterly gut the sacrefical dagger blood drinking, is someone walking around with a bag full of flashes just to weaponize them against certain races (or antags in the case of shadowlings).

You have to consider and think of ways to balance or block power gaming or abusive behavior like that. Even a cooldown on the flash would not be sufficient cause i tried that with the daggers, and people just walked around with a bag of daggers and swapped off.

I am unsure how to get around this all off the top of my head at this time, Besides doing some kind of Diminishing Return on how much damage can be done at any given time.

Tayyyyyyy commented:
Hmm okay, I didn't think of that...however, I doubt this will be a valid powergaming strategy, since there is a RNG roll prob(40) for even dealing damage with a flash on an unprotected character. So with your latest PR with a value of 6 darksight for vulp/taj, taking into account probability, that's only an expected additional damage of `(6 - 2) * 0.4 = 1.6` damage per flash (the additional damage probably needs to be increased tbh). At that point you probably could've just punched them to death, or stabbed them in the eyes with a screw driver, what's the point of carrying a bunch of flashes to blind them?

Aurorablade commented:
Even with RNG, while a good deterrent in some cases and you point of punching being inefficient, people would still do it.
Flashes can stun, make you drop items, and make your screen flash white for a bit, so its pretty much a good thing to use regardless. You add eye damage with a scale to it people will go ham. Yes i realize this is my paranoia but i am jaded by some experiences and first hands, So apology there.

Again one avenue to look at is adding a cap on how much damage a eye could take from a flash in a given amount of time.

Tayyyyyyy commented:
> Again one avenue to look at is adding a cap on how much damage a eye could take from a flash in a given amount of time.

Okay, so the base damage of a flash is 1, and additional damage of (6 - 2), multiplied by 0.4, is an average of 2 damage per flash, which would mean that it would take 15 flashes to completely blind you. I'm not sure why anyone would go through that much effort to just blind someone? At that point i feel like they should be rewarded for going through so much trouble to achieve so little...

Aurorablade commented:
I wonder the same thing and still haven't figured out why people do it, we need a research study.

KingPhilipIII commented:
This will need space law changes on handling prisoners accordingly. There are already very flash happy officers that like to flash you for any kind of infraction(And often it's valid) but given the number of species to be affected by this I feel like there should be a note to restrain from excessive flashes.

Tayyyyyyy commented:
I don't know about that. I'll do some testing to see how much more severe a cell flash is, but Nanotrasen is a human corporation and probably wouldnt change standard operating procedure just to prevent some easily cured eye damage. The brig phys will just have to step up her game. Part of dealing with aliens is that they're alien and thus the station will not be perfectly suited to them.

KasparoVy commented:
A criminal's a criminal haha, you asked for it man

KingPhilipIII commented:
I know, but it falls in the vein of “Don’t best prisoners” in terms of not causing excessive and unneeded harm. I mean, they make specifications on species that don’t breathe oxygen

deep-fried-faceplant commented:
Per average arrest, I'd say prisoners get flashed about two or three times. Maybe once or twice in the initial arrest, and once when taking off the cuffs. I don't see any reason to change SecSOP for a little added damage.

However, this PR got me thinking: Maybe flashes should do more damage to races with darksight if the flash happens in darkness. In a maintenance tunnel with very little light, a flash would be disorienting enough to even stun the target. It would add a situational counter for darksight races, but not without counterplay (just keep a light on you).

And on a tangent, wearing mesons and getting flashed should amplify the damage. I don't know much about mesons or quantum physics, but goggles that let you see through walls through the magic of mesons sound like they'd be susceptible to sudden flashes of light.

Tayyyyyyy commented:
> However, this PR got me thinking: Maybe flashes should do more damage to races with darksight if the flash happens in darkness. In a maintenance tunnel with very little light, a flash would be disorienting enough to even stun the target. It would add a situational counter for darksight races, but not without counterplay (just keep a light on you).

This is a genius idea that would solve the problem of the additional damage being kind of pathetic without guaranteeing blindness if sec decides to flash your cell. I'll look into implementing this!!

KingPhilipIII commented:
NV goggles should have some major flash damage amplification too.

Tayyyyyyy commented:
> NV goggles should have some major flash damage amplification too.

They already do! Part of why the whole getting free 8 darksight was so controversial

KingPhilipIII commented:
Oh they do? Huh didn’t know that.

Tayyyyyyy commented:
Alrighty, I did @deep-fried-faceplant's idea and also removed the RNG of flashes and just converted it to a multiplier instead. Here's what the new damage looks like, which depends on both how brightly lit the tile is, and how much dark view the eyes have:

![capture](https://user-images.githubusercontent.com/26497062/34464582-bea93dea-ee42-11e7-9c38-f7fc73d09320.PNG)

KasparoVy commented:
Hell yeah, now that's more like it.
😎 My eyes are ready.

r3valkyrie commented:
> The issue i have here, and i have seen this before i had to utterly gut the sacrefical dagger blood drinking, is someone walking around with a bag full of flashes just to weaponize them against certain races (or antags in the case of shadowlings).
> 
> You have to consider and think of ways to balance or block power gaming or abusive behavior like that. Even a cooldown on the flash would not be sufficient cause i tried that with the daggers, and people just walked around with a bag of daggers and swapped off.

So we can't use known racial weaknesses against others because you consider it powergaming? Okay, so anyone who:

- Uses any form of light-based attacks against shadowlings.
- Sprays Gray's with water/extinguisher.
- Removes Vox internals.
- Uses Plant-B-Gone against Diona.
- Uses EMPs against IPCs.
- Uses fire against Drasks.

-is powergaming.

I disagree. The biological weaknesses of each species would be well-known from an in-character perspective. In regards to game balance, if a race is going to have built-in night vision, it would be fair to make that race vulnerable to the same tactics you'd use against regular night vision goggles. In this case, flashes.

davidchan13 commented:
To hop on what @r3valkyrie was saying, this isn't that big of a nerf either since welding protection blocks flashes as well meaning that hardsuit helmets, welding masks, welding helmets, welding goggles, sunglasses, SecHUDs and welding implants would all completely negate this downside, the implants themselves being exceptionally cheap and require a very a low research level to create (they are usually one of the first implants unlocked)

deep-fried-faceplant commented:
Flashes are easy to counter, the damage is noticable but not crippling, security can just use their baton while brigging sensitive criminals, and it adds some form of racial vulnerability in line with the other racial benefits and their drawbacks. And, because I can play the realism and RP card, I'm going to and say that this is great for adding a little bit of mechanical fluff to the game.

Vivalas commented:
Isn't weaponizing it the point though? Like, that's what Grey's vulnerability to water is based on.

Rumiluntti commented:
Why is there even a discussion about the potential of people actually using the racial weakness against them?
Isn't that like, the point of this whole thing. They have a racial bonus of seeing in the dark but they get fucked by flashes.
Literally no reason to change flashes or their use because it happens to be the weakness of some races

Aurorablade commented:
Rumi i could care less about the weakness against them in moderation or appropriately...i do care about people loading up on flashes to be a dick.

KingPhilipIII commented:
You’re wanting to preemptively nerfing something because people might “be a dick” and utilize a racial weakness to the fullest extent they can to kill or disable a target or enemy. 

Not a good way to live your life. We might as well make burn damage to shadowlings from being exposed to light cap out, and cap toxin damage dooms receive from weed killer, etc.

davidchan13 commented:
@KingPhilipIII Or make EMPs only stun IPCs and not delimb them.

tigercat2000 commented:
This seems pretty good- the only real concern I see is the effect this will have on sec in regards to cell flashers. However, if that becomes a problem, we can just disable damage on those anyways. For the most part, security doesn't really need to hand-flash people, when they can just taze them or stunbaton them instead, especially given that flashes confuse and make them annoying to latch on to. Additionally, as with any other abuse, if someone goes around flashing taj or vulp to intentionally destroy their eyes, that's a good way for someone to get banned for being a cunt.
